go/pkg/libproxy/multiplexed.go

```diff
@@ -14,29 +14,28 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
-const (
-	maxBufferSize = 65536
-)
+const defaultWindowSize = 65536
 
 type windowState struct {
 	current uint64
 	allowed uint64
+	max     uint64
 }
 
 func (w *windowState) String() string {
-	return fmt.Sprintf("current %d, allowed %d", w.current, w.allowed)
+	return fmt.Sprintf("current %d, allowed %d, max %d", w.current, w.allowed, w.max)
 }
 
 func (w *windowState) size() int {
 	return int(w.allowed - w.current)
 }
 
 func (w *windowState) isAlmostClosed() bool {
-	return w.size() < maxBufferSize/2
+	return w.size() < int(w.max/2)
 }
 
 func (w *windowState) advance() {
-	w.allowed = w.current + uint64(maxBufferSize)
+	w.allowed = w.current + w.max
```

Comment:
I wonder how long it takes to overlap?

Reply:
I think if it went at (a very optimistic) 8Gbit/sec = 1 GiB/sec = `2 ** 30` bytes per second, I think it will overflow to `0` in `2 ** (64-30)` seconds which is > 500 years
```python
>>> 2**(64-30) / 60 / 60 / 24 / 365
544.7700781329274
``` 
I hope we'll have replaced the code by then! (As an aside, I'm interested in any other multiplexing libraries we could use. There was a SPDY implementation but I think it's deprecated. Perhaps HTTP/2 can do what we want? Or maybe even send packets over gRPC? I'm not sure if the overheads would be too much though.)